src/zftools/metagraph.pyx

```diff
@@ -31,14 +31,22 @@ from cpython.mem cimport (
 from zftools.fastqueue cimport FastQueueForBFS
 
 
+def verify_graph_is_compatible(sage_graph):
+    if sage_graph.is_directed():
+        raise ValueError("This is a directed graph.  Currently, directed graph zero forcing is not implemented.")
```

Comment:
Let's change this to raise a `NotImplementedError` instead, for a bit more clarity.

Reply:
Cool.  I didn't even know about `NotImplementedError`!  I don't know about differentiating them.  That would require us to _a priori_ identify all possible reasons a metagraph subclass might decide that it can't instantiate on a certain graph.  But I guess we could just create an “other” subclass of `NotImplementedError`?  I guess the advantage to having something like `DirectedGraphError` is that then when `metagraph.__init__()` throws an error, the calling function (which is, say, `zero_forcing_number()`) can respond by printing some appropriate error message, or maybe also/instead branching according to which type of error was thrown so as to move on to a more appropriate subclass.